lua/barbar/buffer.lua

```diff
@@ -145,9 +146,7 @@ function buffer.get_unique_names(buffer_numbers)
   local update_computed_names = function()
     computed_names = map(function(buffer_number)
       local name = buffer.get_name(buffer_number, depth)
-      local parts = split(name, separator)
-      local computed_name = table_concat(list.slice_from_end(parts, depth), separator)
-      return computed_name
+      return fs.slice_parts_from_end(name, depth)
```

Comment:
This pattern is why `slice_parts_from_end` exists; splitting and slicing was quite common in path disambiguation.